sycl/include/sycl/ext/intel/esimd/math.hpp

```diff
@@ -403,6 +403,42 @@ __ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_COND, sin, sin)
 /// Absolute error: \c 0.0008 or less for the range [-32767*pi, 32767*pi].
 __ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_COND, cos, cos)
 
+template <class T = double, int N, class Sat = saturation_off_tag,
+          class = std::enable_if_t<detail::is_generic_floating_point_v<T> &&
+                                   (sizeof(T) > 4)>>
+__ESIMD_API simd<double, N> sqrt(simd<T, N> src, Sat sat = {}) {
+  return sqrt_ieee(src, sat);
+}
+
+/** Scalar version.                                                       */
+template <class T, class Sat = saturation_off_tag,
+          class = std::enable_if_t<detail::is_generic_floating_point_v<T> &&
+                                   (sizeof(T) > 4)>>
+__ESIMD_API double sqrt(T src, Sat sat = {}) {
+  return sqrt_ieee(src, sat);
+}
+
+template <class T, int N, class Sat = saturation_off_tag,
+          class = std::enable_if_t<detail::is_generic_floating_point_v<T> &&
+                                   (sizeof(T) > 4)>>
+__ESIMD_API simd<double, N> rsqrt(simd<T, N> src, Sat sat = {}) {
+  if constexpr (std::is_same_v<Sat, saturation_off_tag>)
+    return 1. / sqrt(src);
+  else
+    return esimd::saturate<double>(1. / sqrt(src));
+}
+
+/** Scalar version.                                                       */
+template <class T, class Sat = saturation_off_tag,
+          class = std::enable_if_t<detail::is_generic_floating_point_v<T> &&
+                                   (sizeof(T) > 4)>>
+__ESIMD_API double rsqrt(T src, Sat sat = {}) {
```

Comment:
Should it really be such complex here? Can't we keep it simple like this?
```suggestion
/** Scalar version.                                                       */
__ESIMD_API double rsqrt(double src, Sat sat = {}) {
```

Reply:
is it because we always want the template argument `T` to be the 1st tparam?
If Yes, it still can be simpler. Let me finalize the idea here tomorrow.

Reply:
Yes, it is better. Thank you.